lib/matplotlib/image.py

```diff
@@ -162,203 +210,158 @@ def changed(self):
+                    # numbers as transparent.
+                    output = np.empty((out_height, out_width), dtype=A.dtype)
+                    output[...] = -100.0
+                else:
+                    output = np.zeros((out_height, out_width), dtype=A.dtype)
 
-        return im, xmin, ymin, dxintv, dyintv, sx, sy
+                alpha = 1.0
+            elif len(A.shape) == 3:
+                # Always convert to RGBA, even if only RGB input
+                if A.shape[2] == 3:
+                    A = _rgb_to_rgba(A)
+                elif A.shape[2] != 4:
+                    raise ValueError("Invalid dimensions, got %s" % (A.shape,))
 
-    @staticmethod
-    def _get_rotate_and_skew_transform(x1, y1, x2, y2, x3, y3):
-        """
-        Retuen a transform that does
-         (x1, y1) -> (x1, y1)
-         (x2, y2) -> (x2, y2)
-         (x2, y1) -> (x3, y3)
-
-        It was intended to derive a skew transform that preserve the
-        lower-left corner (x1, y1) and top-right corner(x2,y2), but
-        change the lower-right-corner(x2, y1) to a new position
-        (x3, y3).
-        """
-        tr1 = mtransforms.Affine2D()
-        tr1.translate(-x1, -y1)
-        x2a, y2a = tr1.transform_point((x2, y2))
-        x3a, y3a = tr1.transform_point((x3, y3))
+                output = np.zeros((out_height, out_width, 4), dtype=A.dtype)
 
-        inv_mat = 1. / (x2a*y3a-y2a*x3a) * np.mat([[y3a, -y2a], [-x3a, x2a]])
+                alpha = self.get_alpha()
+                if alpha is None:
+                    alpha = 1.0
+            else:
+                raise ValueError("Invalid dimensions, got %s" % (A.shape,))
+
+            _image.resample(
+                A, output, t, _interpd_[self.get_interpolation()],
+                self.get_resample(), alpha,
+                self.get_filternorm() or 0.0, self.get_filterrad() or 0.0)
+
+            output = self.to_rgba(output, bytes=True, norm=False)
+
+            # Apply alpha *after* if the input was greyscale
+            if len(A.shape) == 2:
+                alpha = self.get_alpha()
+                if alpha is not None and alpha != 1.0:
+                    alpha_channel = output[:, :, 3]
+                    alpha_channel[:] = np.asarray(
+                        np.asarray(alpha_channel, np.float) * alpha, np.uint8)
+        else:
+            if self._imcache is None:
+                self._imcache = self.to_rgba(A, bytes=True, norm=(A.ndim == 2))
+            output = self._imcache
 
-        a, b = (inv_mat * np.mat([[x2a], [x2a]])).flat
-        c, d = (inv_mat * np.mat([[y2a], [0]])).flat
+            # Subset the input image to only the part that will be
+            # displayed
+            subset = TransformedBbox(
+                clip_bbox, t0.frozen().inverted()).frozen()
+            output = output[
+                int(max(subset.ymin, 0)):int(min(subset.ymax + 1, A.shape[0])),
+                int(max(subset.xmin, 0)):int(min(subset.xmax + 1, A.shape[1]))]
```

Comment:
Actually, just to clarify -- in this case it doesn't matter in practice since this is the "don't resample" code path, and A.shape == output.shape at this point.  But I agree `output.shape` is clearer than `A.shape` here and will change.